File: app/src/main/java/libertypassage/com/app/widgets/CustomCheckBox.java
package libertypassage.com.app.widgets;

import android.annotation.SuppressLint;
import android.content.Context;
import android.util.AttributeSet;
import android.widget.CheckBox;
import libertypassage.com.app.R;



@SuppressLint("AppCompatCustomView")
public class CustomCheckBox extends CheckBox {


    public CustomCheckBox(Context context, AttributeSet attrs) {
        super(context, attrs);
        //setButtonDrawable(new StateListDrawable());
    }
    @Override
    public void setChecked(boolean t){
        if(t)
        {
            this.setBackgroundResource(R.drawable.check_box_mark);
        }
        else
        {
            this.setBackgroundResource(R.drawable.check_box);
        }
        super.setChecked(t);
    }
}

File: app/src/main/java/libertypassage/com/corporate/utilities/CustomCheckBox.kt
package libertypassage.com.corporate.utilities

import android.annotation.SuppressLint
import android.content.Context
import android.util.AttributeSet
import android.widget.CheckBox
import libertypassage.com.corporate.R


@SuppressLint("AppCompatCustomView")
class CustomCheckBox(context: Context?, attrs: AttributeSet?) : CheckBox(context, attrs) {
    override fun setChecked(t: Boolean) {
        if (t) {
            setBackgroundResource(R.drawable.check_box_mark)
        } else {
            setBackgroundResource(R.drawable.check_box)
        }
        super.setChecked(t)
    }
}